common/network-common/src/main/java/org/apache/spark/network/TransportContext.java

```diff
@@ -187,7 +190,11 @@ public TransportChannelHandler initializePipeline(
       RpcHandler channelRpcHandler) {
     try {
       TransportChannelHandler channelHandler = createChannelHandler(channel, channelRpcHandler);
-      ChannelPipeline pipeline = channel.pipeline()
+      ChannelPipeline pipeline = channel.pipeline();
+      if (nettyLogger.getLoggingHandler() != null) {
```

Comment:
When does this become `null`?

Reply:
When the log level is neither DEBUG nor TRACE:
https://github.com/apache/spark/blob/c5a05d8425f6fa14b34f81469730b43403bdddd7/common/network-common/src/main/java/org/apache/spark/network/util/NettyLogger.java#L58